From f79e216cbf55058c2809511e0568843ca336e0b6 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 20 Aug 2024 18:41:25 +0530
Subject: [PATCH] Enable android mandated drivers to fix vintf check failure

Android15 mandates BUG_ON_DATA_CORRUPTION and SCHED_DEBUG drivers.
vintf check failure seen if those drivers are not enabled.

Tests done:
- Android release build
- Android boot in GVT-d and BM config
- adb reboot

Tracked-On: OAM-123601
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 .../lts2023-chromium/x86_64_defconfig         | 42 ++++++++++++++++---
 1 file changed, 36 insertions(+), 6 deletions(-)

diff --git a/groups/kernel/gmin64/config-lts/lts2023-chromium/x86_64_defconfig b/groups/kernel/gmin64/config-lts/lts2023-chromium/x86_64_defconfig
index 64103e7..290023c 100644
--- a/groups/kernel/gmin64/config-lts/lts2023-chromium/x86_64_defconfig
+++ b/groups/kernel/gmin64/config-lts/lts2023-chromium/x86_64_defconfig
@@ -77,6 +77,7 @@ CONFIG_GENERIC_IRQ_MATRIX_ALLOCATOR=y
 CONFIG_GENERIC_IRQ_RESERVATION_MODE=y
 CONFIG_IRQ_FORCED_THREADING=y
 CONFIG_SPARSE_IRQ=y
+# CONFIG_GENERIC_IRQ_DEBUGFS is not set
 # end of IRQ subsystem
 
 CONFIG_CLOCKSOURCE_WATCHDOG=y
@@ -167,6 +168,7 @@ CONFIG_IKCONFIG_PROC=y
 # CONFIG_IKHEADERS is not set
 CONFIG_LOG_BUF_SHIFT=18
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=12
+# CONFIG_PRINTK_INDEX is not set
 CONFIG_HAVE_UNSTABLE_SCHED_CLOCK=y
 
 #
@@ -333,6 +335,7 @@ CONFIG_X86_MPPARSE=y
 CONFIG_X86_INTEL_LPSS=y
 # CONFIG_X86_AMD_PLATFORM_DEVICE is not set
 CONFIG_IOSF_MBI=y
+# CONFIG_IOSF_MBI_DEBUG is not set
 CONFIG_X86_SUPPORTS_MEMORY_FAILURE=y
 CONFIG_SCHED_OMIT_FRAME_POINTER=y
 CONFIG_HYPERVISOR_GUEST=y
@@ -386,6 +389,7 @@ CONFIG_X86_MCE=y
 CONFIG_X86_MCE_INTEL=y
 # CONFIG_X86_MCE_AMD is not set
 CONFIG_X86_MCE_THRESHOLD=y
+# CONFIG_X86_MCE_INJECT is not set
 
 #
 # Performance monitoring
@@ -406,6 +410,7 @@ CONFIG_X86_MSR=y
 CONFIG_X86_CPUID=y
 # CONFIG_X86_5LEVEL is not set
 CONFIG_X86_DIRECT_GBPAGES=y
+# CONFIG_X86_CPA_STATISTICS is not set
 # CONFIG_NUMA is not set
 CONFIG_ARCH_SPARSEMEM_ENABLE=y
 CONFIG_ARCH_SPARSEMEM_DEFAULT=y
@@ -545,6 +550,7 @@ CONFIG_ACPI_CONTAINER=y
 CONFIG_ACPI_HOTPLUG_IOAPIC=y
 # CONFIG_ACPI_SBS is not set
 # CONFIG_ACPI_HED is not set
+# CONFIG_ACPI_CUSTOM_METHOD is not set
 # CONFIG_ACPI_BGRT is not set
 # CONFIG_ACPI_REDUCED_HARDWARE_ONLY is not set
 # CONFIG_ACPI_NFIT is not set
@@ -553,6 +559,7 @@ CONFIG_HAVE_ACPI_APEI_NMI=y
 CONFIG_ACPI_APEI=y
 # CONFIG_ACPI_APEI_GHES is not set
 # CONFIG_ACPI_APEI_PCIEAER is not set
+# CONFIG_ACPI_APEI_EINJ is not set
 # CONFIG_ACPI_APEI_ERST_DEBUG is not set
 # CONFIG_ACPI_DPTF is not set
 # CONFIG_ACPI_CONFIGFS is not set
@@ -782,6 +789,7 @@ CONFIG_ARCH_HAS_STRICT_MODULE_RWX=y
 CONFIG_STRICT_MODULE_RWX=y
 CONFIG_HAVE_ARCH_PREL32_RELOCATIONS=y
 CONFIG_ARCH_USE_MEMREMAP_PROT=y
+# CONFIG_LOCK_EVENT_COUNTS is not set
 CONFIG_ARCH_HAS_MEM_ENCRYPT=y
 CONFIG_HAVE_STATIC_CALL=y
 CONFIG_HAVE_STATIC_CALL_INLINE=y
@@ -798,6 +806,7 @@ CONFIG_ARCH_HAS_NONLEAF_PMD_YOUNG=y
 #
 # GCOV-based kernel profiling
 #
+# CONFIG_GCOV_KERNEL is not set
 CONFIG_ARCH_HAS_GCOV_PROFILE_ALL=y
 # end of GCOV-based kernel profiling
 
@@ -810,6 +819,7 @@ CONFIG_FUNCTION_ALIGNMENT=16
 CONFIG_RT_MUTEXES=y
 CONFIG_BASE_SMALL=0
 CONFIG_MODULES=y
+# CONFIG_MODULE_DEBUG is not set
 # CONFIG_MODULE_FORCE_LOAD is not set
 CONFIG_MODULE_UNLOAD=y
 # CONFIG_MODULE_FORCE_UNLOAD is not set
@@ -835,6 +845,7 @@ CONFIG_BLK_DEV_BSGLIB=y
 # CONFIG_BLK_DEV_INTEGRITY is not set
 # CONFIG_BLK_DEV_ZONED is not set
 # CONFIG_BLK_WBT is not set
+CONFIG_BLK_DEBUG_FS=y
 # CONFIG_BLK_SED_OPAL is not set
 CONFIG_BLK_INLINE_ENCRYPTION=y
 CONFIG_BLK_INLINE_ENCRYPTION_FALLBACK=y
@@ -998,6 +1009,7 @@ CONFIG_HAVE_SETUP_PER_CPU_AREA=y
 CONFIG_CLEANCACHE=y
 CONFIG_CMA=y
 # CONFIG_CMA_DEBUG is not set
+# CONFIG_CMA_DEBUGFS is not set
 # CONFIG_CMA_SYSFS is not set
 CONFIG_CMA_AREAS=16
 CONFIG_GENERIC_EARLY_IOREMAP=y
@@ -1014,10 +1026,7 @@ CONFIG_ARCH_USES_HIGH_VMA_FLAGS=y
 CONFIG_ARCH_HAS_PKEYS=y
 CONFIG_VM_EVENT_COUNTERS=y
 # CONFIG_PERCPU_STATS is not set
-
-#
-# GUP_TEST needs to have DEBUG_FS enabled
-#
+# CONFIG_GUP_TEST is not set
 # CONFIG_DMAPOOL_TEST is not set
 CONFIG_ARCH_HAS_PTE_SPECIAL=y
 CONFIG_MEMFD_CREATE=y
@@ -1371,6 +1380,7 @@ CONFIG_NF_DEFRAG_IPV6=y
 # CONFIG_TIPC is not set
 # CONFIG_ATM is not set
 CONFIG_L2TP=y
+# CONFIG_L2TP_DEBUGFS is not set
 # CONFIG_L2TP_V3 is not set
 # CONFIG_BRIDGE is not set
 # CONFIG_NET_DSA is not set
@@ -1383,6 +1393,7 @@ CONFIG_VLAN_8021Q=y
 # CONFIG_LAPB is not set
 # CONFIG_PHONET is not set
 CONFIG_6LOWPAN=y
+# CONFIG_6LOWPAN_DEBUGFS is not set
 CONFIG_6LOWPAN_NHC=y
 CONFIG_6LOWPAN_NHC_DEST=y
 CONFIG_6LOWPAN_NHC_FRAGMENT=y
@@ -1543,6 +1554,7 @@ CONFIG_BT_LE_L2CAP_ECRED=y
 # CONFIG_BT_LEDS is not set
 # CONFIG_BT_MSFTEXT is not set
 # CONFIG_BT_AOSPEXT is not set
+CONFIG_BT_DEBUGFS=y
 # CONFIG_BT_SELFTEST is not set
 # CONFIG_BT_FEATURE_DEBUG is not set
 
@@ -1588,6 +1600,7 @@ CONFIG_CFG80211=m
 CONFIG_CFG80211_REQUIRE_SIGNED_REGDB=y
 CONFIG_CFG80211_USE_KERNEL_REGDB_KEYS=y
 CONFIG_CFG80211_DEFAULT_PS=y
+# CONFIG_CFG80211_DEBUGFS is not set
 CONFIG_CFG80211_CRDA_SUPPORT=y
 # CONFIG_CFG80211_WEXT is not set
 CONFIG_LIB80211=m
@@ -1599,6 +1612,7 @@ CONFIG_MAC80211_RC_DEFAULT_MINSTREL=y
 CONFIG_MAC80211_RC_DEFAULT="minstrel_ht"
 # CONFIG_MAC80211_MESH is not set
 CONFIG_MAC80211_LEDS=y
+# CONFIG_MAC80211_DEBUGFS is not set
 # CONFIG_MAC80211_MESSAGE_TRACING is not set
 # CONFIG_MAC80211_DEBUG_MENU is not set
 CONFIG_MAC80211_STA_HASH_MAX_SIZE=0
@@ -1839,6 +1853,7 @@ CONFIG_ZRAM_DEF_COMP_LZORLE=y
 CONFIG_ZRAM_DEF_COMP="lzo-rle"
 # CONFIG_ZRAM_WRITEBACK is not set
 # CONFIG_ZRAM_TRACK_ENTRY_ACTIME is not set
+# CONFIG_ZRAM_MEMORY_TRACKING is not set
 # CONFIG_ZRAM_MULTI_COMP is not set
 CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_LOOP_MIN_COUNT=16
@@ -2607,6 +2622,7 @@ CONFIG_IEEE802154_DRIVERS=y
 
 # CONFIG_VMXNET3 is not set
 # CONFIG_FUJITSU_ES is not set
+# CONFIG_NETDEVSIM is not set
 CONFIG_NET_FAILOVER=y
 # CONFIG_ISDN is not set
 
@@ -3897,6 +3913,7 @@ CONFIG_MEDIA_PCI_SUPPORT=y
 # CONFIG_VIDEO_TW5864 is not set
 # CONFIG_VIDEO_TW68 is not set
 # CONFIG_VIDEO_TW686X is not set
+# CONFIG_VIDEO_ZORAN is not set
 # CONFIG_VIDEO_IPU3_CIO2 is not set
 # CONFIG_INTEL_VSC is not set
 # CONFIG_IPU_BRIDGE is not set
@@ -4577,6 +4594,7 @@ CONFIG_SND_DRIVERS=y
 # CONFIG_SND_PCSP is not set
 CONFIG_SND_DUMMY=m
 # CONFIG_SND_ALOOP is not set
+# CONFIG_SND_PCMTEST is not set
 # CONFIG_SND_VIRMIDI is not set
 # CONFIG_SND_MTPAV is not set
 # CONFIG_SND_SERIAL_U16550 is not set
@@ -5439,6 +5457,7 @@ CONFIG_NOP_USB_XCEIV=y
 CONFIG_USB_GADGET=y
 # CONFIG_USB_GADGET_DEBUG is not set
 # CONFIG_USB_GADGET_DEBUG_FILES is not set
+# CONFIG_USB_GADGET_DEBUG_FS is not set
 CONFIG_USB_GADGET_VBUS_DRAW=2
 CONFIG_USB_GADGET_STORAGE_NUM_BUFFERS=2
 # CONFIG_U_SERIAL_CONSOLE is not set
@@ -5817,6 +5836,7 @@ CONFIG_DW_DMAC_PCI=y
 # DMABUF options
 #
 CONFIG_SYNC_FILE=y
+# CONFIG_SW_SYNC is not set
 # CONFIG_UDMABUF is not set
 # CONFIG_DMABUF_MOVE_NOTIFY is not set
 # CONFIG_DMABUF_DEBUG is not set
@@ -6073,6 +6093,7 @@ CONFIG_IOMMU_SUPPORT=y
 #
 # end of Generic IOMMU Pagetable Support
 
+# CONFIG_IOMMU_DEBUGFS is not set
 # CONFIG_IOMMU_DEFAULT_DMA_STRICT is not set
 CONFIG_IOMMU_DEFAULT_DMA_LAZY=y
 # CONFIG_IOMMU_DEFAULT_PASSTHROUGH is not set
@@ -7172,7 +7193,7 @@ CONFIG_CC_HAS_ZERO_CALL_USED_REGS=y
 # Hardening of kernel data structures
 #
 CONFIG_LIST_HARDENED=y
-# CONFIG_BUG_ON_DATA_CORRUPTION is not set
+CONFIG_BUG_ON_DATA_CORRUPTION=y
 # end of Hardening of kernel data structures
 
 CONFIG_CC_HAS_RANDSTRUCT=y
@@ -7539,6 +7560,7 @@ CONFIG_SWIOTLB=y
 # CONFIG_SWIOTLB_DYNAMIC is not set
 # CONFIG_DMA_CMA is not set
 # CONFIG_DMA_API_DEBUG is not set
+# CONFIG_DMA_MAP_BENCHMARK is not set
 CONFIG_SGL_ALLOC=y
 CONFIG_IOMMU_HELPER=y
 CONFIG_CHECK_SIGNATURE=y
@@ -7620,7 +7642,10 @@ CONFIG_MAGIC_SYSRQ=y
 CONFIG_MAGIC_SYSRQ_DEFAULT_ENABLE=0x1
 CONFIG_MAGIC_SYSRQ_SERIAL=y
 CONFIG_MAGIC_SYSRQ_SERIAL_SEQUENCE=""
-# CONFIG_DEBUG_FS is not set
+CONFIG_DEBUG_FS=y
+CONFIG_DEBUG_FS_ALLOW_ALL=y
+# CONFIG_DEBUG_FS_DISALLOW_MOUNT is not set
+# CONFIG_DEBUG_FS_ALLOW_NONE is not set
 CONFIG_HAVE_ARCH_KGDB=y
 # CONFIG_KGDB is not set
 CONFIG_ARCH_HAS_UBSAN_SANITIZE_ALL=y
@@ -7654,10 +7679,12 @@ CONFIG_SLUB_DEBUG=y
 CONFIG_ARCH_HAS_DEBUG_WX=y
 # CONFIG_DEBUG_WX is not set
 CONFIG_GENERIC_PTDUMP=y
+# CONFIG_PTDUMP_DEBUGFS is not set
 CONFIG_HAVE_DEBUG_KMEMLEAK=y
 # CONFIG_DEBUG_KMEMLEAK is not set
 # CONFIG_PER_VMA_LOCK_STATS is not set
 # CONFIG_DEBUG_OBJECTS is not set
+# CONFIG_SHRINKER_DEBUG is not set
 # CONFIG_DEBUG_STACK_USAGE is not set
 # CONFIG_SCHED_STACK_END_CHECK is not set
 CONFIG_ARCH_HAS_DEBUG_VM_PGTABLE=y
@@ -7707,6 +7734,7 @@ CONFIG_HARDLOCKUP_CHECK_TIMESTAMP=y
 #
 # Scheduler Debugging
 #
+CONFIG_SCHED_DEBUG=y
 CONFIG_SCHED_INFO=y
 CONFIG_SCHEDSTATS=y
 # end of Scheduler Debugging
@@ -7846,6 +7874,7 @@ CONFIG_IO_DELAY_0X80=y
 # CONFIG_IO_DELAY_0XED is not set
 # CONFIG_IO_DELAY_UDELAY is not set
 # CONFIG_IO_DELAY_NONE is not set
+# CONFIG_DEBUG_BOOT_PARAMS is not set
 # CONFIG_CPA_DEBUG is not set
 # CONFIG_DEBUG_ENTRY is not set
 # CONFIG_DEBUG_NMI_SELFTEST is not set
@@ -7867,6 +7896,7 @@ CONFIG_CC_HAS_SANCOV_TRACE_PC=y
 # CONFIG_KCOV is not set
 CONFIG_RUNTIME_TESTING_MENU=y
 # CONFIG_TEST_DHRY is not set
+# CONFIG_LKDTM is not set
 # CONFIG_TEST_MIN_HEAP is not set
 # CONFIG_TEST_DIV64 is not set
 # CONFIG_BACKTRACE_SELF_TEST is not set
-- 
2.45.2

